crates/matrix-sdk-base/src/rooms/normal.rs

```diff
@@ -627,25 +643,32 @@ impl Room {
 
         let own_user_id = self.own_user_id().as_str();
 
-        let (heroes, guessed_num_members): (Vec<String>, _) = if !summary.heroes_names.is_empty() {
-            // Straightforward path: pass through the heroes names, don't give a guess of
-            // the number of members.
-            (summary.heroes_names, None)
-        } else if !summary.heroes_user_ids.is_empty() {
-            // Use the heroes, if available.
-            let heroes = summary.heroes_user_ids;
-
-            let mut names = Vec::with_capacity(heroes.len());
-            for user_id in heroes {
-                if user_id == own_user_id {
-                    continue;
-                }
-                if let Some(member) = self.get_member(&user_id).await? {
-                    names.push(member.name().to_owned());
+        let (heroes, guessed_num_members): (Vec<String>, _) = if !summary.room_heroes.is_empty() {
+            // TODO: Fallback to the local part of the user_id if the display name is
+            // missing?
```

Comment:
This seems to make sense to me, so long as we know for sure that the hero has a loaded profile.

Reply:
How about something simpler like:

- iterate over all the heroes
- if hero == own user id, ignore it
- if hero has a name, use the name
- otherwise try to get a room member for the hero's user_id, and call `display_name()` (which will do the right thing: either use the display name from the member event, or use the localpart)

This way we can avoid the inner `if !hero_names.is_empty()` extra case handling below.

Reply:
> which will do the right thing

Ah, ha! That sounds like a good course of action for computing the room name.